portal/src/AppRoot.tsx

```diff
@@ -1,28 +1,24 @@
 import React, { useMemo } from "react";
-import { Routes, Route, useParams } from "react-router-dom";
-import { makeEnvironment } from "./graphql/adminapi/relay";
-import AppContext from "./AppContext";
-import AppScreen from "./graphql/portal/AppScreen";
+import { Routes, Route, useParams, Navigate } from "react-router-dom";
+import { ApolloProvider } from "@apollo/client";
+import { makeClient } from "./graphql/adminapi/apollo";
 import ScreenLayout from "./ScreenLayout";
-
-const DummyScreen: React.FC = function DummyScreen() {
-  return <p>This is dummy</p>;
-};
+import UsersScreen from "./graphql/adminapi/UsersScreen";
 
 const AppRoot: React.FC = function AppRoot() {
   const { appID } = useParams();
-  const environment = useMemo(() => {
-    return makeEnvironment(appID);
+  const client = useMemo(() => {
+    return makeClient(appID);
   }, [appID]);
   return (
-    <AppContext.Provider value={environment}>
+    <ApolloProvider client={client}>
```

Comment:
It seems this `ApolloProvider` (for admin API) is within another `ApolloProvider` (for portal backend API), can we call portal backend API using `useQuery` hook in the children components?

Reply:
Of course!

```typescript
import { client } from "../portal/apollo";

function MyComponent() {
  // Use client from current context
  const obj1 = useQuery(someAdminAPIQuery);
  // Use the provided client
  const obj2 = useQuery(somePortalQuery, { client });
}
```